### PR TITLE
Staging project 2

### PR DESCRIPTION
**Pull request by JaydenLiang**

### CODE REVIEW

**Review by Joel-Cripps — APPROVED** [review]

File: core/autoscale-core.ts [hunk 1]
```diff
@@ -485,58 +485,54 @@ export abstract class Autoscale implements AutoscaleCore {
...
-                                    this.proxy.logForError(
-                                        'unable to send Autoscale notifications.',
-                                        err
-                                    );
-                                });
-                        })
-                        .catch(err => {
-                            this.proxy.logForError('handling unhealthy vm failed.', err);
-                        });
+        const vmHandler = async function(vm: VirtualMachine, self: Autoscale): Promise<void> {
```
Comment (jamie-pate) [inline core/autoscale-core.ts:488]: why not use a lambda to close over `this` ?

`const vmHandler = async (vm:VirtualMachine): Promise<void> => { ... }`

File: fortigate-autoscale/package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     ],
     "scripts": {
         "test": "echo \"Error: no test specified\" && exit 1",
-        "build": "../node_modules/.bin/tsc -b"
+        "build": "npx tsc -b"
```
Comment (jamie-pate) [inline fortigate-autoscale/package.json:13]: https://docs.npmjs.com/cli/v7/using-npm/scripts#path

tsc should already be added to your path, why do you need npx?

File: package.json [hunk 3]
```diff
@@ -9,12 +9,12 @@
         "build": "npm-run-all pre-build clear-dist compile",
         "build-artifacts": "npm-run-all build make-dist",
         "bundle": "cd fortigate-autoscale && npm unlink --no-save @fortinet/autoscale-core && cd .. && npm pack ./fortigate-autoscale",
-        "clear-dist": "rimraf */dist",
-        "compile": "tsc --build",
+        "clear-dist": "npx rimraf dist/ && npx rimraf fortigate-autoscale/dist/",
```
Comment (jamie-pate) [inline package.json:12]: why do you need npx? the bin directory should already be in the path when running scripts

**Review by jamie-pate — CHANGES_REQUESTED** [review]

File: core/autoscale-core.ts (hunk 1 above)
Comment (JaydenLiang) [reply core/autoscale-core.ts]: done

File: fortigate-autoscale/package.json (hunk 2 above)
Comment (JaydenLiang) [reply fortigate-autoscale/package.json:13]: done

File: package.json (hunk 3 above)
Comment (JaydenLiang) [reply package.json]: done

File: fortigate-autoscale/package.json (hunk 2 above)
Comment (jamie-pate) [reply fortigate-autoscale/package.json:13]: is npx really needed?

**Review by jamie-pate — CHANGES_REQUESTED** [review]

**Pull request merged**